server/utils/device.js

```diff
@@ -68,8 +68,48 @@ function getDeviceFeature(device, category, type) {
   return null;
 }
 
+const matchFeature = (features, feature) => {
+  return features.findIndex((f) => f.external_id === feature.external_id);
+};
+
+const matchParam = (params, param) => {
+  return params.findIndex((p) => p.name === param.name && p.value === param.value);
+};
+
+/**
+ * @description Compare newDevice with exisingDevice on features and parameters
+ * to tell if device can be updated.
+ *
+ * @param {Object} newDevice - New device.
+ * @param {Object} existingDevice - Existing device.
+ * @returns {boolean} Is new device is different on feature or parameters?
+ * @example
+ * isUpdatable({}, {})
+ */
+function isUpdatable(newDevice, existingDevice = {}) {
```

Comment:
what would make a device updatable or not ? Maybe the description is not self-explanatory, I don't understand why some device could be updated and some couldn't :)

Reply:
You're right, the name is not explicit enough,
this method can be used to "compare" device already stored in Gladys with newly discovored one.
It compares features on external_id and params on name and value.
If any is different, it returns true, so this can be used to display "update" button in discovery tab.

I'm open to any remarks on this point.

Reply:
hasDeviceChanged?

Reply:
I agree, with more detailled comment to explain "changed on what".
Thanks ;)